pkg/resources/aws/aws.go

```diff
@@ -1179,47 +1179,34 @@ func ListAutoScalingGroups(cloud fi.Cloud, clusterName string) ([]*resources.Res
 	return resourceTrackers, nil
 }
 
-// FindAutoScalingLaunchTemplateConfigurations finds any launch configurations which reference the security groups
-func FindAutoScalingLaunchTemplateConfigurations(cloud fi.Cloud, securityGroups sets.String) ([]*resources.Resource, error) {
-	var list []*resources.Resource
-
+// FindAutoScalingLaunchTemplates finds any launch configurations which reference the security groups
+func FindAutoScalingLaunchTemplates(cloud fi.Cloud, clusterName string, securityGroups sets.String) ([]*resources.Resource, error) {
 	c, ok := cloud.(awsup.AWSCloud)
 	if !ok {
 		return nil, errors.New("expected a aws cloud provider")
 	}
 	klog.V(2).Infof("Finding all Autoscaling LaunchTemplates associated to security groups")
 
-	resp, err := c.EC2().DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{MaxResults: fi.Int64(100)})
+	output, err := c.EC2().DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:KubernetesCluster"),
+				Values: []*string{aws.String(clusterName)},
+			},
+		},
+	})
 	if err != nil {
-		return list, nil
+		return nil, nil
 	}
 
-	for _, x := range resp.LaunchTemplates {
-		// @step: grab the actual launch template
-		req, err := c.EC2().DescribeLaunchTemplateVersions(&ec2.DescribeLaunchTemplateVersionsInput{
-			LaunchTemplateName: x.LaunchTemplateName,
+	var list []*resources.Resource
+	for _, lt := range output.LaunchTemplates {
+		list = append(list, &resources.Resource{
```

Comment:
Where is it filtering for referencing `securityGroups`? I don't see where `securityGroups` is referenced.

Reply:
Just forgot to remove it. Should be addressed in the next commit.